src/main/java/baseball/BallStrikeCalculator.java

```diff
@@ -0,0 +1,32 @@
+package baseball;
+
+import static baseball.ConstValue.BASEBALL_MAX_LENGTH;
+
+import java.util.Arrays;
+import java.util.stream.IntStream;
+
+public class BallStrikeCalculator {
+
+    // 스트라이크와 볼 갯수를 Score 객체로 반환
+    public Score countStrikeAndBallToScore(int[] userAnswer, int[] computerAnswer) {
+        int strikes = countStrike(userAnswer, computerAnswer);
+        int balls = countBall(userAnswer, computerAnswer);
+
+        return new Score(strikes, balls);
+    }
+
+    // 스트라이크 숫자 갯수 반환
+    public int countStrike(int[] userAnswer, int[] computerAnswer) {
+        return (int) IntStream.range(0, BASEBALL_MAX_LENGTH)
+                .filter(i -> userAnswer[i] == computerAnswer[i])
+                .count();
+    }
+
+    // 볼 숫자 갯수 반환
+    public int countBall(int[] userAnswer, int[] computerAnswer) {
+        return (int) IntStream.range(0, BASEBALL_MAX_LENGTH)
+                .filter(i -> userAnswer[i] != computerAnswer[i])
+                .filter(i -> Arrays.stream(computerAnswer).anyMatch(x -> x == userAnswer[i]))
+                .count();
+    }
```

Comment:
볼 개수를 계산하는 메서드네요

filter 조건이 두개가 들어가 있는데 위치가 다르면서 같은 번호를 서로 포함하고 있는지 확인하기 위해 이렇게 구현되어 있는 것 같아요

다만 아래의 필터는 스트림을 하나 새로 열고 있는데 만약 제가 위에 말씀드린 것처럼 ```int[]```가 아닌 ```List<Integer>```를 사용했다면 다음과 같이 바뀔 수 있지 않을까 싶습니다 

```suggestion
    public int countBall(List<Integer> userAnswer, List<Integer> computerAnswer) {
        return (int) IntStream.range(0, BASEBALL_MAX_LENGTH)
                .filter(i -> userAnswer.get(i) != computerAnswer.get(i))
                .filter(computerAnswer.contains(userAnswer.get(i)
                )
                .count();
    }
```